### PR TITLE
Vertical obstructed

### PR DESCRIPTION
**Pull request by EGimenoS**
Approaching is_obstructed? method in a modular way, I first defined vertical_obstructed? that returns true in case a vertical move from x_position, y_position to x,y is obstructed by another piece on the board.

Defined severeal tests as well, for both cases moving from top to bottom and from bottom to top. the method takes into account that thepiece calling the method should not trigger true.

### CODE REVIEW

File: app/models/piece.rb [hunk 1]
```diff
@@ -34,4 +34,17 @@ def horizontal?(x, y)
   def vertical?(x, y)
     (x_position - x).zero? && (y_position - y).abs > 0
   end
+  # method to check is the piece move is vertically obstructed
+  
+  def vertical_obstructed?(x,y)
```
Comment (lasley) [inline app/models/piece.rb]: Always put spaces after your commas: `(x, y)`

File: app/models/piece.rb [hunk 2]
```diff
@@ -34,4 +34,17 @@ def horizontal?(x, y)
...
+    
+    (y_position...y).each do |y|
```
Comment (lasley) [inline app/models/piece.rb]: This seems inefficient. What if you leveraged the power of ActiveRecord instead & searched for pieces on `x_position = piece.x_position` and `y_position in (y_position...y)`

File: spec/models/piece_spec.rb [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
-  describe 'common restrictions to all pieces' do
-    it 'should move inside of the limits of the board' do
-      dummy_king = Piece.create(type: 'King', color: 'black', image: '', x_position: 4, y_position: 0, is_captured: false, game_id: 0)
-      expect(dummy_king.inside_limits?(4, 4)).to eql(true)
-      expect(dummy_king.inside_limits?(8, 4)).to eql(false)
+    
+
+    RSpec.describe Piece, type: :model do
+      before(:each) do
+        @dummy_rook = Piece.create(type: 'Rook', color: 'white', image: '', x_position: 0, y_position: 4, is_captured: false, game_id: 0)
```
Comment (lasley) [inline spec/models/piece_spec.rb]: image

File: spec/models/piece_spec.rb [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
     end
 
-    it 'should make a move to a square different than the current one' do
-      dummy_king = Piece.create(type: 'King', color: 'black', image: '', x_position: 4, y_position: 0, is_captured: false, game_id: 0)
-      expect(dummy_king.moved?(4, 0)).to eql(false)
-      expect(dummy_king.moved?(4, 1)).to eql(true)
+      describe "obstructions" do
+        it "should return true if the vertical move is blocked" do
+          expect(@dummy_rook.vertical_obstructed?(0,1)).to eql(true)
+        end
```
Comment (lasley) [inline spec/models/piece_spec.rb]: What about non-obstructed?

Comment (EGimenoS) [inline spec/models/piece_spec.rb]: I don't know what you mean here, I wrote 4 test cases, 2 for non blocked and other 2 for blocked

https://github.com/it-is-5pm-somwhere/team-chess-app/blob/0c5478223076d27768961b38dc564e6ab78e3401/spec/models/piece_spec.rb

Comment (saakuffo) [inline spec/models/piece_spec.rb]: So the problem here is that you are manually building the cases and not using Factories. We have FactoryGirl implemented.

Comment (EGimenoS) [inline spec/models/piece_spec.rb]: You are right, I will refactor using factories.

**Comment by EGimenoS** [comment]
@lasley @saakuffo please check and review, I improved verical_obstructed method to be more efficient as Dave pointed,  and I used factories to create pieces to tests.

The significant commit is this one, the last one just fixes a couple of missclicks

https://github.com/it-is-5pm-somwhere/team-chess-app/pull/19/commits/a45bb63809faef099980749a634e650dbb3fc515

**Comment by saakuffo** [comment]
Not passing tests

**Comment by EGimenoS** [comment]
Going to close this PR as it is included in a bigger one for the complete obstructed method and I feel that we are commenting on obsolete code.

**Pull request closed without merging**